src/aura/BGE_TemplateDetails/BGE_TemplateDetailsController.js

```diff
@@ -43,6 +43,27 @@
 
     /******************************** Template Fields Controller Functions *****************************/
 
+
+    /* ***************************************************************
+     * @Description. Updates the selected active fields in the
+     * Template Fields Model to available.
+     *****************************************************************/
+    moveFieldsDown: function (component, event, helper) {
+        var model = component.get('v.model');
+        var selectedRows = component.find("activeTreeGrid").getSelectedRows();
+        model.getTemplateFields().moveFieldsDown(selectedRows);
+    },
+
+    /* ***************************************************************
+     * @Description. Updates the selected active fields in the
+     * Template Fields Model to available.
+     *****************************************************************/
+    moveFieldsUp: function (component, event, helper) {
+        var model = component.get('v.model');
+        var selectedRows = component.find("activeTreeGrid").getSelectedRows();
```

Comment:
@randi274 here, we don't need to pass the Selected Rows, because the model has them already. We can do something similar to:
`moveFieldsUp: function (component, event, helper) {
        var model = component.get('v.model');
        model.getTemplateFields().moveSelectedUp();
    }`

We did a similar logic in the controller method updateToActive:
`updateToActive: function (component, event, helper) {
        var model = component.get('v.model');
        model.getTemplateFields().updateSelectedToActive();
    }`

Reply:
True! I think we were hoping to make use of attributes the tree grid adds, like posInSet, and setSize, such as on line 610 in the helper. Is there a good way around using that logic, or is that something we might need to add to the model?